src/components/Banner/index.tsx

```diff
@@ -0,0 +1,129 @@
+import { isString } from 'lodash'
+import { type ReactNode, useState } from 'react'
+import styled from 'styled-components'
+
+import { THEME } from '../../theme'
+
+export enum BannerType {
+  'CLOSABLE' = 'CLOSABLE',
+  'COLLAPSIBLE' = 'COLLAPSIBLE'
+}
+export enum BannerVersion {
+  'ERROR' = 'ERROR',
+  'VALID' = 'VALID',
+  'WARNING' = 'WARNING'
+}
+
+export type BannerProps = {
+  actionButton: ReactNode
+  isHiddenByDefault: boolean | undefined
+  text: string | ReactNode
+  top: string
+  type: BannerType
+  version: BannerVersion
+}
+
+const colorPalette = (version: BannerVersion) => {
+  let config = {
+    backgroundColor: THEME.color.mediumSeaGreen25,
+    borderColor: THEME.color.mediumSeaGreen,
+    color: THEME.color.mediumSeaGreen
+  }
+  if (version === BannerVersion.ERROR) {
+    config = {
+      backgroundColor: THEME.color.maximumRed15,
+      borderColor: THEME.color.maximumRed,
+      color: THEME.color.maximumRed
+    }
+  } else if (version === BannerVersion.WARNING) {
+    config = {
+      backgroundColor: THEME.color.goldenPoppy25,
+      borderColor: THEME.color.goldenPoppy,
+      color: THEME.color.charcoal
+    }
+  }
+
+  return config
```

Comment:
```suggestion
  if (version === BannerVersion.ERROR) {
    return {
      backgroundColor: THEME.color.maximumRed15,
      borderColor: THEME.color.maximumRed,
      color: THEME.color.maximumRed
    }
  }
  
  if (version === BannerVersion.WARNING) {
    return {
      backgroundColor: THEME.color.goldenPoppy25,
      borderColor: THEME.color.goldenPoppy,
      color: THEME.color.charcoal
    }
  }

  return {
    backgroundColor: THEME.color.mediumSeaGreen25,
    borderColor: THEME.color.mediumSeaGreen,
    color: THEME.color.mediumSeaGreen
  }
```

Je trouve ça plus lisible en early return, wdyt (c'est plutôt du taste) ?

Reply:
Alors au choix, je suis aussi pour l'écrire différemment :
- soit comme dit Loup en l'utilisant comme un util mais alors en le renommant comme une fonction `getWrapperStyle(accent: BannerAccent)` et en le déplaçant dans un fichier `utils.ts`
- soit, comme tu as un enum, tout simplement via une constant `const BANNER_ACCENT_STYLE: Record<BannerAccent, CSSProperties> = {...}` dans un fichier `constants.ts`

Reply:
perso j'aurais plutôt géré ça comme on fait pour les boutons par exemple (et si tu te bases sur une prop `Level` comme j'ai mis en commentaire) 
Avec un switch sur le Level comme ici pour les boutons : https://github.com/MTES-MCT/monitor-ui/blob/main/src/elements/Button.tsx#L73